src/main/java/ch/njol/skript/effects/EffTransform.java

```diff
@@ -0,0 +1,139 @@
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Keywords;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.ExprInput;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.InputSource;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.Variable;
+import ch.njol.skript.lang.parser.ParserInstance;
+import ch.njol.skript.variables.Variables;
+import ch.njol.util.Kleenean;
+import ch.njol.util.Pair;
+import ch.njol.util.StringUtils;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+@Name("Transform List")
+@Description({
+	"Transforms (or 'maps') a list's values using a given expression. This is akin to looping over the list and setting " +
+	"each value to a modified version of itself.",
+	"If the given expression returns a single value, the indices of the list will not change. If the expression returns " +
+	"multiple values, then then indices will be reset as a single index cannot contain multiple values.",
+	"Only variable lists can be transformed with this effect. For other lists, see the transform expression."
+})
```

Comment:
```suggestion
@Description({
	"Transforms (or 'maps') a list's values using a given expression. This is akin to looping over the list and setting " +
	"each value to a modified version of itself.",
	"Applies the given function to each element in the list, replacing the original element with the function's result.",
	"If the given expression returns a single value, the indices of the list will not change. If the expression returns " +
	"multiple values, then then indices will be reset as a single index cannot contain multiple values.",
	"Only variable lists can be transformed with this effect. For other lists, see the transform expression."
})
```

Reply:
i like it, though i'll replace it with `expression`